File: src/main/java/ee/leola/kassa/models/TransactionProduct.java
package ee.leola.kassa.models;

import jakarta.persistence.Entity;
import jakarta.persistence.OneToOne;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class TransactionProduct extends Model {

  @NotNull private String name;

  @NotNull private BigDecimal price = BigDecimal.ZERO;

  private Integer quantity;

  @OneToOne private Product product;

  public TransactionProduct() {}

  public TransactionProduct(Product product, Integer quantity) {
    setQuantity(quantity);
    setName(product.getName());
    setPrice(product.getPrice());
    setProduct(product);
  }
}

File: src/main/kotlin/ee/leola/kassa/models/TransactionProduct.kt
package ee.leola.kassa.models

import jakarta.persistence.Entity
import jakarta.persistence.OneToOne
import jakarta.validation.constraints.NotNull
import java.math.BigDecimal

@Entity
class TransactionProduct(
    val name: @NotNull String,
    val price: @NotNull BigDecimal = BigDecimal.ZERO,
    val quantity: Int,
    @OneToOne
    val product: Product,
) : Model()
